packages/story-editor/src/app/media/utils/useMediaInfo.js

```diff
@@ -0,0 +1,249 @@
+        const result = JSON.parse(
+          await mediaInfo.analyzeData(getSize, readChunk)
+        );
+
+        const normalizedResult = result.media.track.reduce(
+          (acc, track) => {
+            if (track['@type'] === 'General') {
+              acc.fileSize = Number(track.FileSize);
+              acc.format = track.Format.toLowerCase().replace('mpeg-4', 'mp4');
+              acc.frameRate = Number(Number(track.FrameRate).toFixed(0));
+              acc.codec = track.CodecID?.trim();
+            }
+
+            if (track['@type'] === 'Image' || track['@type'] === 'Video') {
+              acc.width = Number(track.Width);
+              acc.height = Number(track.Height);
+              acc.colorSpace = track.ColorSpace; // Maybe useful in the future.
+            }
+
+            if (track['@type'] === 'Video') {
+              acc.duration = Number(track.Duration);
+              acc.videoCodec = track.Format.toLowerCase();
+            }
+
+            if (track['@type'] === 'Audio') {
+              acc.audioCodec = track.Format.toLowerCase();
+            }
+
+            return acc;
+          },
+          {
+            mimeType: file.type,
+          }
+        );
+
+        normalizedResult.isMuted = !normalizedResult.audioCodec;
+
+        mediaInfo.close();
+
+        return normalizedResult;
+      } catch (err) {
+        // eslint-disable-next-line no-console -- We want to surface this error.
+        console.error(err);
+
+        trackError('mediainfo', err.message);
+
+        return null;
+      } finally {
+        trackTiming();
+      }
+    },
+    [mediainfoUrl]
+  );
+
+  const isConsideredOptimized = useCallback(
+    /**
+     * Determines whether a video file is to be considered optimized.
+     *
+     * Checks things like dimensions, file size, mime type, and codecs.
+     *
+     * @todo Allow WebM with VP9 once Safari catches up.
+     * @param {Resource} resource Resource object.
+     * @param {File} file File object.
+     * @return {Promise<boolean>} Whether the file meets optimization criteria.
+     */
+    async (resource, file) => {
+      // This should never happen, but just in case.
+      if (resource.isOptimized) {
+        return true;
+      }
+
+      // Short-circuit for non-matching mime types.
+      if (resource.mimeType !== 'video/mp4') {
+        return false;
+      }
+
+      // Placeholders are the size of the canvas, so account for that when
+      // checking the dimensions.
+      if (
+        !resource.isPlaceholder &&
+        resource.width * resource.height >
+          MEDIA_VIDEO_DIMENSIONS_THRESHOLD.WIDTH *
+            MEDIA_VIDEO_DIMENSIONS_THRESHOLD.HEIGHT
+      ) {
+        return false;
+      }
+
+      const fileInfo = await getFileInfo(file);
+
+      if (!fileInfo) {
+        return false;
+      }
+
+      const hasSmallFileSize =
+        fileInfo.fileSize < MEDIA_VIDEO_FILE_SIZE_THRESHOLD;
```

Comment:
```suggestion
      const threshold = fileInfo.duration * 266666.667;
      const hasSmallFileSize = fileInfo.fileSize < threshold;
```

A bit of testing, this seems to work. For a 45 second video, it 11.1MB, which seems right.

Reply:
See https://github.com/GoogleForCreators/web-stories-wp/pull/11454#issuecomment-1174038967

There are some disadvantages with this method.